### PR TITLE
Block generation [OLB-16]

### PR DESCRIPTION
**Pull request by jliyazhu12**
Added block generation and texture.
Missing adjustments to the animated texture for the active/inactive states.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -13,3 +13,10 @@ run
 .classpath
 .project
 .settings/
+/bin/
```
Comment (jamestjw) [inline .gitignore]: `/bin/`? What is this supposed to be? Did you mean to write `./bin` or 'bin/' instead/

**Review by jamestjw — CHANGES_REQUESTED** [review]
Everything else is cool :)

File: .gitignore (hunk 1 above)
Comment (jliyazhu12) [reply .gitignore]: I only added to .gitignore starting from src/generated, the rest was already there.

Comment (jamestjw) [reply .gitignore]: that is unfortunately not true, here is a screenshot of what the gitignore file looks like in the `main` branch 
image

Comment (jamestjw) [reply .gitignore]: @jliyazhu12 if you never meant to add `bin` in there,  I suppose you could just scrap that line entirely.

Comment (jliyazhu12) [reply .gitignore]: I honestly have no idea how that was added. When I try to remove it, it lists bin/ as "untracked files", so I modified it to that.

**Comment by jamestjw** [comment]
Okay unfortunately the CI pipeline doesn't run on forked repos, could you clone the repository and make a pull request from a feature branch instead? @jliyazhu12

**Pull request closed without merging**